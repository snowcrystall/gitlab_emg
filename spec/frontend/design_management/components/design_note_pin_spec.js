import { shallowMount } from '@vue/test-utils';
import DesignNotePin from '~/design_management/components/design_note_pin.vue';

describe('Design note pin component', () => {
  let wrapper;

  function createComponent(propsData = {}) {
    wrapper = shallowMount(DesignNotePin, {
      propsData: {
        position: {
          left: '10px',
          top: '10px',
        },
        ...propsData,
      },
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  it('should match the snapshot of note without index', () => {
    createComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match the snapshot of note with index', () => {
    createComponent({ label: 1 });
    expect(wrapper.element).toMatchSnapshot();
  });
});
