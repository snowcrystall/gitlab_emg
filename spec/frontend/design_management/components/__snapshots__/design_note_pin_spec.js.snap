// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design note pin component should match the snapshot of note with index 1`] = `
<button
  aria-label="Comment '1' position"
  class="gl-absolute gl-display-flex gl-align-items-center gl-justify-content-center gl-font-lg gl-outline-0! js-image-badge badge badge-pill"
  style="left: 10px; top: 10px;"
  type="button"
>
  
    1
  
</button>
`;

exports[`Design note pin component should match the snapshot of note without index 1`] = `
<button
  aria-label="Comment form position"
  class="gl-absolute gl-display-flex gl-align-items-center gl-justify-content-center gl-font-lg gl-outline-0! btn-transparent comment-indicator gl-p-0"
  style="left: 10px; top: 10px;"
  type="button"
>
  <gl-icon-stub
    name="image-comment-dark"
    size="24"
  />
</button>
`;
